src/main/kotlin/org/stellar/walletsdk/Data.kt

```diff
@@ -1,7 +1,45 @@
 package org.stellar.walletsdk
 
+data class AccountSigner(val address: String, val weight: Int)
+
+data class AccountThreshold(val low: Int, val medium: Int, val high: Int)
```

Comment:
Wallet SDK users will use these data types for inputs, so it might be more convenient (and clear) to have them directly in the Wallet SDK. @JakeUrban, are you suggesting that we should rather document which data type to use from the Java SDK?

![image](https://user-images.githubusercontent.com/7346473/193831947-9558f0df-c4ad-44d1-b3a6-3449c382e481.png)

Reply:
If we are introducing Java types, we need to keep in mind that the Kotlin dev experience might be affected. For example, named arguments will not be supported.

![image](https://user-images.githubusercontent.com/7346473/193882488-49a528fb-86bf-4fe7-bb3e-09f096953055.png)

Also, where do we draw the line of what we use from Java SDK? Will it be only types that match exactly (`Thresholds`) or anything that comes close (`Signer` is a class in Java SDK with more params than we need)?

What are the benefits of coupling Java SDK and Wallet SDK? I'd vote for adding duplicate types if it improves user experience, but my opinion might be biased (I've been working on this for a while).

Reply:
Your point about named arguments is compelling. My concern is that the wallet SDK becomes a collection of very simple wrappers that deliver little value.

I think the value of the wallet SDK is abstracting processes, not wrapping Java SDK 1-liners. For example, `Wallet.authenticate()` for SEP-10 is a huge improvement. But this is not as powerful and could cause confusion for folks who are trying to understand the difference between the Java SDK and the Wallet SDK.

My vote would be to start with the highest level abstractions in the Wallet SDK, because that is what is most valuable. And then over time, we can work our way down to lower abstractions if we think it makes sense.

I also think that there will be many instances where it makes sense to make a change to the Java SDK, vs. adding an abstraction on top of it with the Wallet SDK. For example, I think we should make a change to the Java SDK's `AccountResponse.Signer` to remove the additional argument (key format).

Reply:
From a Kotlin developer's point of view, I felt it does not bother to use Java style classes. The named arguments are compelling. But the work-around is to use the [Lombok's builder ](https://projectlombok.org/features/Builder) pattern. Unfortunately, our Horizon SDK does not use that. 

I don't have strong preference. While using native Kotlin classes is cool, using native Java classes does not bother me either.

Reply:
I think if we decide to wrap Java objects, we probably should consistently wrap all of the classes to avoid developer confusion. 

Hmm... not sure if it is very helpful to wrap the Java classes with the effort spent.

Reply:
I forgot to update this thread on what we discussed offline. Wallet SDK will be a standalone library and not a companion to the Stellar Java SDK. The question here was whether we should replicate things we already have in Java SDK. Now we know the answer: yes. We'll add everything needed for the Wallet SDK to operate independently from the Java SDK. :crossed_fingers: